src/game/level.c

```diff
@@ -671,7 +731,190 @@ bool Level_Load(int level_num)
         }
     }
 
-    g_GameFlow.levels[level_num].secrets = GetSecretCount();
+    Level_CalculateStats();
+    g_GameFlow.levels[level_num].secrets = m_LevelSecrets;
+    g_GameFlow.levels[level_num].pickups = m_LevelPickups;
+    g_GameFlow.levels[level_num].kills = m_LevelKillables;
+    Memory_FreePointer(&m_FloorArray);
 
     return ret;
 }
+
+static void Level_CalculateStats()
+{
+    // Clear old values
+    m_LevelPickups = 0;
+    m_LevelKillables = 0;
+    m_LevelSecrets = 0;
+    memset(&m_Pierres, 0, sizeof(m_Pierres));
+    memset(&m_Mummies, 0, sizeof(m_Mummies));
+    memset(&m_Pods, 0, sizeof(m_Pods));
+    memset(&m_Bats, 0, sizeof(m_Bats));
+    memset(&m_Statues, 0, sizeof(m_Statues));
+
+    if (m_UninitItemCount) {
+        if (m_UninitItemCount > MAX_ITEMS) {
+            Shell_ExitSystem(
+                "Level_GetPickupCount(): Too Many g_Items being Loaded!!");
+            return;
+        }
+
+        for (int i = 0; i < m_UninitItemCount; i++) {
+            ITEM_INFO *item = &g_Items[i];
+
+            if (item->object_number < 0 || item->object_number >= O_NUMBER_OF) {
+                Shell_ExitSystemFmt(
+                    "Level_GetPickupCount(): Bad Object number (%d) on Item %d",
+                    item->object_number, i);
+            }
+
+            // Calculate number of pickups in a level
+            for (int j = 0; m_PickupObjs[j] != NO_ITEM; j++) {
+                if (item->object_number == m_PickupObjs[j]) {
+                    m_LevelPickups++;
+                }
+            }
+            // Spawn pickups on death
+            if (item->object_number == O_SKATEKID) {
+                m_LevelPickups += SKATEKID_ITEMS;
+            } else if (item->object_number == O_COWBOY) {
+                m_LevelPickups += COWBOY_ITEMS;
+            } else if (item->object_number == O_BALDY) {
+                m_LevelPickups += BALDY_ITEMS;
+            }
+
+            // Calculate number of killable objects in a level
+            for (int j = 0; m_KillableObjs[j] != NO_ITEM; j++) {
+                if (item->object_number == m_KillableObjs[j]) {
+                    m_LevelKillables++;
+                }
+            }
+        }
+    }
+
+    // Check triggers for special pickups / killables
+    Level_CalculateTriggers();
+}
+
+static bool Level_FindElement(int16_t elem, int16_t array[], int16_t size)
+{
+    for (int16_t i = 0; i < size; i++) {
+        if (array[i] == elem) {
+            return true;
+        }
+    }
+    return false;
```

Comment:
I don't get the purpose of this function

Reply:
Since there can be multiple triggers for the same object, the item index is stored to make sure the same item isn’t counter multiple times. The idea was to use the array kind of like a hash map. If there’s a better way to do this let me know. My C is rusty and in Python I would have used a dict.

Reply:
Maybe a linked list of found objects?

Reply:
I guess I don't need a key and a value in a dict. I just said that because it's easy to check if a key is already in the dict to avoid adding duplicates. C doesn't have much standard library support with containers. I could put objects in a linked list, but I need to check for duplicates before insertion, or just add everything and remove duplicates after right? If you have a recommendation let me know. I'm honestly not sure how best to implement that in C.